### PR TITLE
Tool voting

### PR DESCRIPTION
**Pull request by kziemianek**
Tools voting functionality
##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines]

##### Affected core subsystem(s)

- backend
- frontend

##### Testing
Tested manually

##### Refers/Fixes
Refs: #706

### CODE REVIEW

**Comment by kziemianek** [comment]
image

**Comment by owocki** [comment]
very excited to see this!

testing notes:
1. i seem to get a 401 unauthorized http response when i try to vote.  instead of just 401ing, the system should prvide an alert telling the user to login.
2. after i vote, i dont seem to be able to undo my vote (eitehr by pressing the button again or by pressing the opposite button)

**Comment by owocki** [comment]
code looks sane.. looking forward to getting this shipped!

**Comment by kziemianek** [comment]
@owocki done :checkered_flag:

**Comment by owocki** [comment]
word; mind resolving the merge conflicts.

@mbeacom you mentioned on slack you had a contribution or two you wanted to make to this PR?  is that still current?

**Comment by mbeacom** [comment]
@owocki Yes, it is! I'll revisit this shortly!  @kziemianek Would you mind resolving the merge conflicts?

File: app/dashboard/migrations/0051_auto_20180401_0947.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Generated by Django 2.0.3 on 2018-04-01 16:47
+
+from django.db import migrations
+
+def insert_tools(apps, schema_editor):
+    Tool = apps.get_model('dashboard', 'Tool')
+
+    #basics
+    issue_explorer = Tool(name="Issue Explorer", category="BASIC", img="v2/images/why-different/code_great.png", description='''A searchable index of all of the funded work available in
```
Comment (kziemianek) [inline app/dashboard/migrations/0051_auto_20180401_0947.py]: For example this is not translated anymore as before, I wonder how it should work. 
I can add to model method _name which will call gettext_lazy.

Comment (owocki) [reply app/dashboard/migrations/0051_auto_20180401_0947.py]: maybe we can just run each user-facing attribute of the model through `gettext_lazy` before it hits the view...


edit: i think youre saying the same thing as me! carry on :)

File: app/dashboard/migrations/0050_tool.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Generated by Django 2.0.3 on 2018-04-01 16:47
```
Comment (mbeacom) [inline app/dashboard/migrations/0050_tool.py]: Migrations will need recreated.

File: app/dashboard/migrations/0051_auto_20180401_0947.py (hunk 1 above)
Comment (mbeacom) [reply app/dashboard/migrations/0051_auto_20180401_0947.py]: @kziemianek You should be able to use `trans` or `blocktrans` template tags in the `toolbox.html` template itself.

Comment (kziemianek) [reply app/dashboard/migrations/0051_auto_20180401_0947.py]: I moved translation code to model, hope it's fine, but it really smells like view layer thing.

Comment (mbeacom) [reply app/dashboard/migrations/0051_auto_20180401_0947.py]: 🙌 That works!

**Comment by mbeacom** [comment]
@kziemianek Mind resolving the conflicts here?

**Comment by kziemianek** [comment]
@mbeacom resolved

File: app/dashboard/migrations/0055_merge_20180417_1455.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 2.0.3 on 2018-04-17 21:55
```
Comment (mbeacom) [inline app/dashboard/migrations/0055_merge_20180417_1455.py]: @kziemianek Mind deleting this merge migration, `0052_auto_20180401_1112.py`, and `0050_tool.py` ? Then move your initial data creation migration out, recreate automatic migrations via `make migrations`, and re-add your data creation migration?

File: app/dashboard/models.py [hunk 4]
```diff
@@ -1121,3 +1123,50 @@ class Meta:
...
+    def i18n_description(self):
+        return _(self.description)
+
+    def i18n_link_copy(self):
+        return _(self.link_copy)
+    
+class ToolVote(models.Model):
+    """Define the vote placed on a tool."""
+
+    profile = models.ForeignKey('dashboard.Profile', related_name='votes', on_delete=models.CASCADE)
```
Comment (mbeacom) [inline app/dashboard/models.py]: 🙌

**Comment by kziemianek** [comment]
@mbeacom now I get it, migrations recreated.

**Pull request merged**